File: minecraft/lib/client/logging.ts
import { z } from 'zod';

export const LoggingFileSchema = z.object({
  id: z.string(),
  sha1: z.string(),
  size: z.number().int(),
  url: z.string(),
});

export type LoggingFile = z.infer<typeof LoggingFileSchema>;

export const LoggingClientSchema = z.object({
  argument: z.string(),
  file: LoggingFileSchema,
  type: z.string(),
});

export type LoggingClient = z.infer<typeof LoggingClientSchema>;

export const LoggingSchema = z.object({ client: LoggingClientSchema });
export type Logging = z.infer<typeof LoggingSchema>;

File: crates/opys-mojang/src/logging.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct LoggingFile {
    pub id: String,
    pub sha1: String,
    pub size: u64,
    pub url: String,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct LoggingClient {
    pub argument: String,
    pub file: LoggingFile,
    #[serde(rename = "type")]
    pub kind: String,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct Logging {
    pub client: LoggingClient,
}
